### ISSUE TITLE
#[forbid(missing_docs)] causes an unresolveable error when expanding a derive macro

### ISSUE THREAD

**Issue by alice-i-cecile** [rust-lang/rust-clippy#8180]
### Summary

When attempting to use the EnumIter derive macro in a crate with #[forbid(missing_docs)], I receive the following error:

allow(missing_docs) incompatible with previous forbid
overruled by previous forbid

The error seems to be somehow caused by the generated code, but I don't see any place where they are explicitly allowing it 🤔

Cross-posted from https://github.com/Peternator7/strum/issues/200, as this seems to be a problem with Clippy.

### Reproducer

I tried this code:

```rust
#![forbid(missing_docs)]
use strum_macros::EnumIter;
 
#[derive(EnumIter)]
enum Foo{
  A,
  B,
}
```

Errors with:
> allow(missing_docs) incompatible with previous forbid
> overruled by previous forbid

The `EnumIter` derive is underlined by RA.

### Version

```text
rustc 1.59.0-nightly (f8abed9ed 2021-12-26)
binary: rustc
commit-hash: f8abed9ed48bace6be0087bcd44ed534e239b8d8
commit-date: 2021-12-26
host: x86_64-pc-windows-msvc
release: 1.59.0-nightly
LLVM version: 13.0.0
```

**Comment by alice-i-cecile** [comment]
For additional context, other derive macros used in this project, including the macro defined [here](https://github.com/Leafwing-Studios/leafwing-input-manager/tree/main/macros) do not cause this issue.

**Comment by Jarcho** [comment]
This is not a clippy issue. Rustc disallows using `#[allow(..)]` after `#[forbid(..)]` on the same lint.
